pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -364,17 +365,67 @@ def _get_tools(self, model_request_parameters: ModelRequestParameters) -> list[T
         return tools or None
 
     def _get_tool_config(
-        self, model_request_parameters: ModelRequestParameters, tools: list[ToolDict] | None
+        self,
+        model_request_parameters: ModelRequestParameters,
+        tools: list[ToolDict] | None,
+        model_settings: GoogleModelSettings,
     ) -> ToolConfigDict | None:
-        if not model_request_parameters.allow_text_output and tools:
-            names: list[str] = []
+        if not tools:
+            return None
+
+        resolved = resolve_tool_choice(model_settings, model_request_parameters)
+
+        if resolved is None:
+            # Default behavior: infer from allow_text_output
+            if not model_request_parameters.allow_text_output:
+                names: list[str] = []
+                for tool in tools:
+                    for function_declaration in tool.get('function_declarations') or []:
+                        if name := function_declaration.get('name'):  # pragma: no branch
+                            names.append(name)
+                return _tool_config(names)
+            return None
+
+        if resolved.mode == 'auto':
+            return ToolConfigDict(
+                function_calling_config=FunctionCallingConfigDict(mode=FunctionCallingConfigMode.AUTO)
```

Comment:
Same issue as above. I'll stop commenting on the cases where the exact same issue shows up for multiple providers